### PR TITLE
Multiple items

### PR DESCRIPTION
**Pull request by LidorWa**

### CODE REVIEW

File: src/app/components/Drawer.jsx [hunk 1]
```diff
@@ -48,13 +48,21 @@ const Select = styled.select`
   color: #7364f0;
   font-size: 16px;
   font-weight: 600;
+
+  :hover{
```
Comment (nir905) [inline src/app/components/Drawer.jsx]: needs to be `cursor: pointer` directly without `hover`

Comment (LidorWa) [reply src/app/components/Drawer.jsx]: Yes, boss!

File: src/modal/components/ItemModal.jsx [hunk 2]
```diff
@@ -40,6 +40,16 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
       }}
       secondaryText={t("cancel")}
       onSecondaryClick={onClose}
+      thirdText={t("add_another_item")}
+      onThirdClick={() => {
+        if (!state.name || state.amount === undefined){
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: Please extract the common code (primary & third click) to another function.
```
onThirdClick={()=>{
  onSaveItem()
   setState({})
}
```

File: src/modal/components/ItemModal.jsx [hunk 3]
```diff
@@ -40,6 +40,16 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
...
+          return;
+        }
+        onSave(state);
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: Also, in react you shouldn't mutate the state. You should create a new object and use the setter to clupdate the state

**Review by nirapx — CHANGES_REQUESTED** [review]

File: src/modal/components/ItemModal.jsx [hunk 4]
```diff
@@ -13,33 +13,53 @@ const Input = styled.input`
   margin-bottom: 32px;
 `;
 
+// const itemAddedModal = styled.div`
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: Remove comments

File: src/modal/components/ItemModal.jsx [hunk 5]
```diff
@@ -13,33 +13,53 @@ const Input = styled.input`
...
+
 const DeleteIcon = styled(BaseDeleteIcon)`
   width: 32px;
   height: 32px;
   color: #f11a1a;
+  cursor: pointer;
 `;
 
+
+
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: It seems you are not using prettier. Please add it or remove one new line

File: src/modal/components/ItemModal.jsx [hunk 6]
```diff
@@ -13,33 +13,53 @@ const Input = styled.input`
...
-        onSave(state);
+        onSaveItem();
         onClose();
       }}
       secondaryText={t("cancel")}
       onSecondaryClick={onClose}
+      thirdText={t("add_another_item")}
+      onThirdClick={() => {
+        onSaveItem();
+        setState({});
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: The initial state is {type: unit} (line 35). After adding the first item - it should keep the same behaviour (default type -unit)

File: src/modal/components/ItemModal.jsx [hunk 7]
```diff
@@ -96,7 +116,6 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
           setState((prev) => ({ ...prev, comment: e.target.value }))
         }
       />
-
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx:97]: Please keep this new line

File: src/shared/components/Modal/Modal.jsx [hunk 8]
```diff
@@ -59,7 +59,6 @@ const CloseIconWrapper = styled.div`
 const Content = styled.div`
   padding: 16px;
 `;
-
```
Comment (nirapx) [inline src/shared/components/Modal/Modal.jsx]: Please keep this new line

**Review by nirapx — CHANGES_REQUESTED** [review]

File: src/modal/components/ItemModal.jsx (hunk 4 above)
Comment (LidorWa) [reply src/modal/components/ItemModal.jsx]: It's part of a feature not yet ready. should i uncomment? remove comments entirely?

Comment (nirapx) [reply src/modal/components/ItemModal.jsx]: Which feature? If it's not used in this PR remove entirely

File: src/modal/components/ItemModal.jsx [hunk 9]
```diff
@@ -13,33 +13,53 @@ const Input = styled.input`
...
-        if (!state.name || state.amount === undefined) {
-          return;
-        }
-        onSave(state);
+        onSaveItem();
         onClose();
       }}
       secondaryText={t("cancel")}
       onSecondaryClick={onClose}
+      thirdText={t("add_another_item")}
```
Comment (nirapx) [inline src/modal/components/ItemModal.jsx]: Third button should be removed in edit mode

**Review by nirapx — CHANGES_REQUESTED** [review]

File: src/modal/components/ItemModal.jsx [hunk 10]
```diff
@@ -26,20 +27,35 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
...
-        if (!state.name || state.amount === undefined) {
-          return;
-        }
-        onSave(state);
+        onSaveItem();
         onClose();
       }}
       secondaryText={t("cancel")}
       onSecondaryClick={onClose}
+      {...(!item.id
```
Comment (nir905) [inline src/modal/components/ItemModal.jsx]:
```
thirdText: item.id &&  t("add_another_item")
onThirdClick: onAddAnotherItemClick,
```
is enough

File: src/shared/components/Modal/Modal.jsx [hunk 11]
```diff
@@ -107,6 +109,11 @@ const Modal = ({
               {secondaryText}
             </SecondaryButton>
           )}
+          {thirdText && (
```
Comment (nir905) [inline src/shared/components/Modal/Modal.jsx]: It doesn't look good
image
try to make it look like this:
image

use this grid in the `Actions` component:
```
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 8px;
```
+ more changes in the buttons style

File: src/modal/components/ItemModal.jsx [hunk 12]
```diff
@@ -26,20 +27,35 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
...
       onClose={onClose}
       title={item.id ? t("update_item") : t("add_item")}
       primaryText={t("save")}
       onPrimaryClick={() => {
-        if (!state.name || state.amount === undefined) {
-          return;
-        }
-        onSave(state);
+        onSaveItem();
         onClose();
```
Comment (nir905) [inline src/modal/components/ItemModal.jsx]: pay attention it closes the modal even if it's not saving anything

File: src/modal/components/ItemModal.jsx [hunk 13]
```diff
@@ -26,20 +27,35 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
...
+  const onSaveItem = () => {
+    if (!state.name || state.amount === undefined) {
+      return;
+    }
+    onSave(state);
+  };
+
+  const onAddAnotherItemClick = () => {
+    onSaveItem();
+    setState({ type: "unit" });
```
Comment (nir905) [inline src/modal/components/ItemModal.jsx]: pay attention it resets the state even if it's not saving anything

File: src/inventory/components/Inventory.jsx [hunk 14]
```diff
@@ -154,7 +154,7 @@ const Inventory = () => {
           ))}
         </Select>
       </SearchWrapper>
-
+      <div>{filteredList.length} {t("products_in_list")}</div>
```
Comment (nir905) [inline src/inventory/components/Inventory.jsx]: should not be seen if there are not items
image

Comment (nir905) [reply src/inventory/components/Inventory.jsx]: and add `margin-bottom: 17px;` to the div

**Review by nir905 — CHANGES_REQUESTED** [review]

File: src/inventory/components/Inventory.jsx (hunk 14 above)
Comment (nir905) [reply src/inventory/components/Inventory.jsx]: you know what? please remove it (and the translations). we need another way to show the number of items..

**Review by nir905 — APPROVED** [review]

**Pull request merged**